tasks/main.py

```diff
@@ -0,0 +1,281 @@
+'''Main task entrypoint'''
+
+import os
+
+from datetime import datetime
+
+from invoke import task, UnexpectedExit
+from stopit import TimeoutException, SignalTimeout as Timeout
+
+from log_utils import get_logger
+from log_utils.remote_logs import (
+    post_output_log, post_build_complete,
+    post_build_error, post_build_timeout)
+from .common import load_dotenv, delta_to_mins_secs
+
+LOGGER = get_logger('MAIN')
+
+TIMEOUT_SECONDS = 20 * 60  # 20 minutes
+
+
+def replace_private_values(text, private_values,
+                           replace_string='[PRIVATE VALUE HIDDEN]'):
+    '''
+    Replaces instances of the strings in `private_values` list
+    with `replace_string`.
+
+    >>> replace_private_values('18F boop Rulez boop', ['boop'])
+    '18F [PRIVATE VALUE HIDDEN] Rulez [PRIVATE VALUE HIDDEN]'
+
+    >>> replace_private_values('18F boop Rulez beep',
+    ...                        ['boop', 'beep'], 'STUFF')
+    '18F STUFF Rulez STUFF'
+    '''
+    for priv_val in private_values:
+        text = text.replace(priv_val, replace_string)
+
+    return text
```

Comment:
if you want to be cool and functional... ;)

`reduce(lambda txt, val: txt.replace(val, replace_string), private_values, text)`

(though this is harder to read than yours imo)

Reply:
I would absolutely do it that way in javascript, but I shy away from the functional stuff in python because of the syntax and silly `lambda` keyword